lua/lean/lsp.lua

```diff
@@ -14,46 +12,51 @@ function lsp.enable(opts)
   require('lspconfig').leanls.setup(opts)
 end
 
+--- Finds the vim.lsp.client object for the Lean 4 server associated to the
+--- given bufnr.
+function lsp.get_lean4_server(bufnr)
+  local lean_client
+  vim.lsp.for_each_buffer_client(bufnr, function (client)
+    if client.name == 'leanls' then lean_client = client end
+  end)
+  return lean_client
+end
+
+--- Finds the vim.lsp.client object for the Lean 3 server associated to the
+--- given bufnr.
+function lsp.get_lean3_server(bufnr)
+  local lean_client
+  vim.lsp.for_each_buffer_client(bufnr, function (client)
+    if client.name == 'lean3ls' then lean_client = client end
+  end)
+  return lean_client
+end
+
 -- Fetch goal state information from the server.
```

Comment:
I think it would be a good idea to mention in the header when functions are async.

Reply:
I've added it to the docstring.  In general, all our LSP functions are async though.